opensquirrel/exporter/quantify_scheduler_exporter.py

```diff
@@ -61,14 +60,23 @@ def visit_bloch_sphere_rotation(self, g: BlochSphereRotation) -> None:
             self.schedule.add(quantify_scheduler_gates.Rz(theta=theta, qubit=self._get_qubit_string(g.qubit)))
             return
 
-        raise _unsupported_gates_exception
+        raise UnsupportedGatesError(
+            f"Cannot export circuit: it contains unsupported gates. The following gate could not be exported: {g}. "
+            "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
+        )
 
     def visit_matrix_gate(self, g: MatrixGate) -> None:
-        raise _unsupported_gates_exception
+        raise UnsupportedGatesError(
+            f"Cannot export circuit: it contains unsupported gates. MatrixGates are not supported. "
+            "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
+        )
 
     def visit_controlled_gate(self, g: ControlledGate) -> None:
         if not isinstance(g.target_gate, BlochSphereRotation):
-            raise _unsupported_gates_exception
+            raise UnsupportedGatesError(
+                f"Cannot export circuit: it contains unsupported gates. The following gate could not be exported: {g}. "
+                "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
+            )
```

Comment:
And from here we would call with `raise UnsupportedGatesError(f"{g}")`, or `str(g)`...